### PR TITLE
Implemented PCIe Topology page

### PR DESCRIPTION
**Pull request by tiwari-nishant**
* PCIe Topology Page Vue2 to Vue3
* Jira Story: https://jsw.ibm.com/browse/PFEBMC-2371

### CODE REVIEW

File: src/views/HardwareStatus/PcieTopology/ResetLinkModal.vue [hunk 1]
```diff
@@ -30,81 +31,95 @@
...
         }}
       </b-button>
     </template>
   </b-modal>
 </template>
-<script>
-import VuelidateMixin from '@/components/Mixins/VuelidateMixin';
-import BVToastMixin from '@/components/Mixins/BVToastMixin';
+<script setup>
+// import VuelidateMixin from '@/components/Mixins/VuelidateMixin';
```
Comment (Nikhil-Ashoka) [inline src/views/HardwareStatus/PcieTopology/ResetLinkModal.vue]: Remove commented code

File: src/views/HardwareStatus/PcieTopology/PcieTopology.vue [hunk 2]
```diff
@@ -5,14 +5,14 @@
       <b-col md="8" xl="6">
         <alert v-if="isInPhypStandby" variant="info" class="mb-4">
           <span>
-            {{ $t('pagePcieTopology.alert') }}
+            {{ $t("pagePcieTopology.alert") }}
```
Comment (Nikhil-Ashoka) [inline src/views/HardwareStatus/PcieTopology/PcieTopology.vue]: Why double quotes? I think the esLint is not working. Look into it.

File: src/store/modules/GlobalStore.js [hunk 3]
```diff
@@ -49,10 +49,18 @@ export const GlobalStore = defineStore('global', {
...
+      state.bootProgress === 'SystemHardwareInitializationComplete' ||
+      state.bootProgress === 'SetupEntered' ||
+      state.bootProgress === 'OSBootStarted' ||
+      state.bootProgress === 'OSRunning',
     isOSRunningGetter: (state) => state.bootProgress === 'OSRunning',
     getIsUtcDisplay: (state) => state.isUtcDisplay,
     safeModeGetter: (state) => state.safeMode,
     serverStatusGetter: (state) => state.serverStatus,
+    isServiceUser: (state) =>
+      state.currentUser?.RoleId === 'OemIBMServiceAgent' || !state.currentUser,
```
Comment (vedangimittal) [inline src/store/modules/GlobalStore.js:65]: state currentUser is neither initialised nor assigned a value anywhere in the file. getCurrentUser() and getAccountService() functions need to be included from older file. Or directly state userPrivilege can be used as well to get the role id of the user.

File: src/views/HardwareStatus/PcieTopology/PcieTopology.vue [hunk 4]
```diff
@@ -5,486 +5,435 @@
...
+const tableIsBusy = computed(() => {
+  if (!globalStore.isInPhypStandby) return false;
+  if (fetched.value == true && globalStore.isInPhypStandby) return false;
+  return true;
+})
+const isInPhypStandby = computed(() => {
+  return globalStore.isInPhypStandby;
+});
+const isServiceUser = computed(() => {
+  return globalStore.isServiceUser;
```
Comment (nabilpandiyalayil) [inline src/views/HardwareStatus/PcieTopology/PcieTopology.vue:349]: This is returning as true for all users(eg: admin)

File: src/views/HardwareStatus/PcieTopology/PcieTopology.vue [hunk 5]
```diff
@@ -5,486 +5,435 @@
...
+import useTableRowExpandComposable from "../../../components/Composables/useTableRowExpandComposable";
+import useSearchFilterComposable from "../../../components/Composables/useSearchFilterComposable";
+import useTableFilter from "../../../components/Composables/useTableFilterComposable";
+import usePaginationComposable from "../../../components/Composables/usePaginationComposable";
+import useLoadingBar from "@/components/Composables/useLoadingBarComposable";
+import { PcieTopologyStore } from "../../../store";
+import { GlobalStore } from "../../../store";
+import eventBus from "@/eventBus";
+import useToast from "@/components/Composables/useToastComposable";
+import { useI18n } from "vue-i18n";
```
Comment (nabilpandiyalayil) [inline src/views/HardwareStatus/PcieTopology/PcieTopology.vue]: use import i18n from '@/i18n';

File: src/views/HardwareStatus/PcieTopology/IdentifyLedsModal.vue [hunk 6]
```diff
@@ -88,93 +90,111 @@
...
 
-export default {
-  mixins: [VuelidateMixin, DataFormatterMixin, BVToastMixin],
-  props: {
-    selectedObj: {
+import { ref,defineProps,watch, computed} from 'vue';
+import { PcieTopologyStore } from '../../../store';
+import useDataFormatterGlobal from '../../../components/Composables/useDataFormatterGlobal';
+import useToast from '@/components/Composables/useToastComposable';
+import { useI18n } from 'vue-i18n';
```
Comment (nabilpandiyalayil) [inline src/views/HardwareStatus/PcieTopology/IdentifyLedsModal.vue]: use import i18n from '@/i18n';

File: src/views/HardwareStatus/PcieTopology/ResetLinkModal.vue [hunk 7]
```diff
@@ -30,81 +31,94 @@
...
   </b-modal>
 </template>
-<script>
-import VuelidateMixin from '@/components/Mixins/VuelidateMixin';
-import BVToastMixin from '@/components/Mixins/BVToastMixin';
+<script setup>
+import { GlobalStore, PcieTopologyStore } from '../../../store';
+import useVuelidate from '@vuelidate/core';
+import useToast from '@/components/Composables/useToastComposable';
+import { useI18n } from 'vue-i18n';
```
Comment (nabilpandiyalayil) [inline src/views/HardwareStatus/PcieTopology/ResetLinkModal.vue]: use import i18n from '@/i18n';

File: src/views/HardwareStatus/PcieTopology/ResetLinkModal.vue [hunk 8]
```diff
@@ -30,81 +31,94 @@
...
+      v$.value.$touch()
+      if (v$.value.$invalid) return;
+      nextTick(() => modal.value.hide());
+      resetConfirm();
+}
+const emitUpdate=defineEmits(['update:openResetModal'])
+function resetConfirm() {
+      confirm.value = false;
+      v$.value.$reset() 
+      emitUpdate('update:openResetModal',false)
```
Comment (nabilpandiyalayil) [inline src/views/HardwareStatus/PcieTopology/ResetLinkModal.vue:99]: Are we using these emits anywhere?

File: src/views/HardwareStatus/PcieTopology/ResetLinkModal.vue [hunk 9]
```diff
@@ -30,81 +31,94 @@
...
+
+}
+function resetLink() {
+  console.log("resetLink fn")
```
Comment (nabilpandiyalayil) [inline src/views/HardwareStatus/PcieTopology/ResetLinkModal.vue]: remove console logs

**Review by nabilpandiyalayil — CHANGES_REQUESTED** [review]

File: src/views/HardwareStatus/PcieTopology/ResetLinkModal.vue (hunk 8 above)
Comment (tiwari-nishant) [reply src/views/HardwareStatus/PcieTopology/ResetLinkModal.vue:99]: yes the root component PCIeToppology uses it to get the updated value

**Review by nabilpandiyalayil — APPROVED** [review]

**Review by vedangimittal — APPROVED** [review]

**Pull request merged**